lib/helper.js

```diff
@@ -16,8 +18,31 @@ function isEmpty(value) {
   return typeof value == 'string' && !value.trim() || typeof value == 'undefined' || value === null;
 }
 
+function encodeKey(metricName, tags) {
+  if (isEmpty(tags)) {
+    return metricName;
+  }
+  // Sort the tags based on their keys to ensure we don't duplicate metrics if the same tags come out of order.
+  var jsonArray = Object.keys(tags).sort().map(key => [key, tags[key]]);
+  var tagsJson = {};
+  jsonArray.forEach(function(tagkv) {
+    tagsJson[tagkv[0]] = tagkv[1];
```

Comment:
This is still depending on object tag ordering.  You need to stringify the array, and not an object built from the array.